File: Easy/219-Contains-Duplicate-II/Solutions.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        hasil = {}
        for i in range(len(nums)):
            duplikat = nums[i]
            if duplikat in hasil and abs(i - hasil[duplikat]) <= k: return True
            hasil[duplikat] = i
        return False

File: Easy/219-Contains-Duplicate-II/Solutions.cpp
class Solution
{
public:
    bool containsNearbyDuplicate(vector<int> &nums, int k)
    {
        unordered_map<int, int> exist;
        for (int i = 0; i < nums.size(); i++)
        {
            int same = nums[i];
            if (exist.find(same) != exist.end() && abs(i - exist[same]) <= k)
                return true;
            exist[same] = i;
        }
        return false;
    }
};